src/renderer/stores/init-user-store.injectable.ts

```diff
@@ -4,20 +4,25 @@
  */
 import { getInjectable } from "@ogre-tools/injectable";
 import userStoreInjectable from "../../common/user-store/user-store.injectable";
+import setupAppPathsInjectable from "../app-paths/setup-app-paths.injectable";
 import { beforeFrameStartsInjectionToken } from "../before-frame-starts/before-frame-starts-injection-token";
 import initDefaultUpdateChannelInjectable from "../vars/default-update-channel/init.injectable";
 
 const initUserStoreInjectable = getInjectable({
   id: "init-user-store",
-  instantiate: (di) => {
-    const userStore = di.inject(userStoreInjectable);
+  instantiate: (di) => ({
+    id: "init-user-store",
+    run: () => {
+      // This must be done here so as to actually only be instantiated after the dependencies are
```

Comment:
Hmm. The smell of temporal dependency is here again :/ This is tech debt, and the proper fix is most probably to make `initDefaultUpdateChannelInjectable` async along with its own dependencies.

Reply:
Nope. It already is async...

That shows that this concern is not correct (at least not fully correct). Furthermore, stating that you want to make it "Async which resolves when complete" is very similar to my previous solution of "InitializableState".

Given that the shape of a `Runnable` is very close to an `Injectable` anyway, (it has an ID, it has a function that gets run at some point). I think it is entirely reasonable to say that the best practice is to just treat it has a way to provide more metadata to an injectable and then say "always inject in the run()".